Unable to unwrap `Option` field of nested struct

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 6b3521f, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 89d1aac.6b3521f
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.6.8-arch1-1 (linux@archlinux) (gcc (GCC) 13.2.1 20230801, GNU ld (GNU Binutils) 2.41.0) #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000
|Processor           |16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P
|Memory              |5.09GB/15.35GB
|                    |
|V executable        |/home/ge/.vlang/v
|V last modified time|2025-03-29 12:43:00
|                    |
|V home dir          |OK, value: /home/ge/.vlang
|VMODULES            |OK, value: /home/ge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/ge
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.13-26-g6b3521f5
|.git/config present |true
|                    |
|cc version          |cc (GCC) 14.2.1 20250207
|gcc version         |gcc (GCC) 14.2.1 20250207
|clang version       |clang version 19.1.7
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.41

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  /tmp/cannot_unwrap_option_field_from_nested_struct.v && /tmp/cannot_unwrap_option_field_from_nested_struct`
```v
struct Foo {
	optional_one ?string
}

struct Bar {
	foo Foo
}

fn main() {
	b := Bar{Foo{
		optional_one: 'hello world'
	}}
	println(b.foo.optional_one)
	x := b.foo.optional_one as string

	assert x == 'hello world'

	// This assert also fails, but with another C error:
	// error: cannot convert 'struct _option_string' to 'struct string'
	assert (b.foo.optional_one as string) == 'hello world'
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/cannot_unwrap_option_field_from_nested_struct.01JQH4PBEZYMPTXF9NW0W4SK6S.tmp.c:5693: error: '{' expected (got ";")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

No C errors, assertions must pass.


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -7554,6 +7554,12 @@ fn (mut g Gen) as_cast(node ast.AsCast) {
 				g.unwrap_option_type(node.expr.typ, '${node.expr.expr.name}.${node.expr.field_name}',
 					node.expr.expr.is_auto_heap())
 				is_optional_ident_var = true
+			} else if node.expr.expr is ast.SelectorExpr && node.expr.typ.has_flag(.option) {
+				if node.expr.expr.expr is ast.Ident {
+					g.unwrap_option_type(node.expr.typ, '${node.expr.expr.expr.name}.${node.expr.expr.field_name}.${node.expr.field_name}',
+						node.expr.expr.expr.is_auto_heap())
+					is_optional_ident_var = true
+				}
 			}
 		}
 		if !is_optional_ident_var {

--- a/vlib/v/tests/options/option_embed_field_test.v
+++ b/vlib/v/tests/options/option_embed_field_test.v
@@ -1,0 +1,18 @@
+struct Foo {
+	optional_one ?string
+}
+
+struct Bar {
+	foo Foo
+}
+
+fn test_main() {
+	b := Bar{Foo{
+		optional_one: 'hello world'
+	}}
+	println(b.foo.optional_one)
+	x := b.foo.optional_one as string
+
+	assert x == 'hello world'
+	assert (b.foo.optional_one as string) == 'hello world'
+}

